### ISSUE TITLE
C# data flow analysis incorrectly reports range variable of the from clause as captured

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#1291]
Given the following method body:
```c#
    public static void Main()
    {
        int y = 1;
        {
            int x = 2;
            var f2 = from a in new[] { 1 } select a + y;
            var f3 = from a in new[] { 1 } where x > 0 select a;
        }
    }
```

```model.AnalyzeDataFlow(memberBody).Captured```

returns 4 variables:
```
[0]          Local System.Int32 y       Microsoft.CodeAnalysis.CSharp.Symbol {Microsoft.CodeAnalysis.CSharp.Symbols.SourceLocalSymbol.LocalWithInitializer}
[1]          Local System.Int32 x       Microsoft.CodeAnalysis.CSharp.Symbol {Microsoft.CodeAnalysis.CSharp.Symbols.SourceLocalSymbol.LocalWithInitializer}
[2]          RangeVariable ? a            Microsoft.CodeAnalysis.CSharp.Symbol {Microsoft.CodeAnalysis.CSharp.Symbols.RangeVariableSymbol}
[3]          RangeVariable ? a            Microsoft.CodeAnalysis.CSharp.Symbol {Microsoft.CodeAnalysis.CSharp.Symbols.RangeVariableSymbol}
```
However, the range variable a is not lifted to closure in this case. It’s translated to parameters of the selector lambdas.

Equivalent code in VB
```VB
    Sub F()
        Dim y As Integer = 1
        Do
            Dim x As Integer = 2
            Dim f2 = From a In {1} Select a + y
            Dim f3 = From a In {1} Where x > 0 Select a
        Loop
    End Sub
```
Only reports x and y as captured. 

Seems like C# got it wrong.

**Comment by gafter** [comment]
See also #1312 - VB has the "opposite" problem.

**Comment by gafter** [comment]
@tmat You asked me to tell you when this is fixed. It is fixed.

**Comment by gafter** [comment]
@tmat Can you please "Verify" that this is fixed?
